### PR TITLE
Update the agones-allocator doc to recommend using cert-manager

### PR DESCRIPTION
**Pull request by pooneh-m**
As the first step towards https://github.com/googleforgames/agones/issues/1461, updating the documentation.

### CODE REVIEW

File: site/content/en/docs/Advanced/allocator-service.md [hunk 1]
```diff
@@ -33,36 +33,48 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
...
+EOF
 
-# After having the TLS certificates ready, run the following command to store the certificate as a Kubernetes TLS secret.
-kubectl create secret --save-config=true tls allocator-tls -n agones-system --key=${TLS_KEY_FILE} --cert=${TLS_CERT_FILE} --dry-run -o yaml | kubectl apply -f -
+EXTERNAL_IP=`kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'`
 
-# Optional: Add the TLS signing CA to allocator-tls-ca
-TLS_CERT_FILE_VALUE=`cat ${TLS_CERT_FILE} | base64 -w 0`
-kubectl get secret allocator-tls-ca -o json -n agones-system | jq '.data["tls-ca.crt"]="'${TLS_CERT_FILE_VALUE}'"' | kubectl apply -f -
+# Create a Certificate with IPs for the allocator-tls secret
```
Comment (aLekSer) [inline site/content/en/docs/Advanced/allocator-service.md:56]: nit: I see that only one IP address `EXTERNAL_IP` could contain.
```suggestion
# Create a Certificate with IP for the allocator-tls secret
```

Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md]: Thanks! Done.

File: site/content/en/docs/Advanced/allocator-service.md [hunk 2]
```diff
@@ -33,36 +33,48 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
 
 ## Server TLS certificate
 
-Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate, including using CA. The following provides an example of generating a self-signed certificate using openssl and storing it in allocator-tls Kubernetes secret.
+Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using cert-manager.io solution for cluster level certificate management.
```
Comment (markmandel) [inline site/content/en/docs/Advanced/allocator-service.md:36]: Nit: cert-manager.io should be a link

File: site/content/en/docs/Advanced/allocator-service.md [hunk 3]
```diff
@@ -33,36 +33,48 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
...
+  commonName: ${EXTERNAL_IP}
+  ipAddresses:
+    - ${EXTERNAL_IP}
+  secretName: allocator-tls
+  issuerRef:
+    name: selfsigned
+    kind: ClusterIssuer
+EOF
+
+# Optional: Store the secret ca.crt in a file and add the TLS ca.crt to the allocator-tls-ca Secret
```
Comment (markmandel) [inline site/content/en/docs/Advanced/allocator-service.md:73]: We might want to be clear about why this is optional?

I'm assuming this is optional if you want to be able to connect to this endpoint from a client?

Also, if you don't do this step, what happens?

File: site/content/en/docs/Advanced/allocator-service.md [hunk 4]
```diff
@@ -33,36 +33,48 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
...
-Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate, including using CA. The following provides an example of generating a self-signed certificate using openssl and storing it in allocator-tls Kubernetes secret.
+Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using cert-manager.io solution for cluster level certificate management. 
 
-```bash
-#!/bin/bash
-EXTERNAL_IP=`kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'`
-
-TLS_KEY_FILE=tls.key
-TLS_CERT_FILE=tls.crt
+In order to use cert-manager solution, first, [install cert-manager](https://cert-manager.io/docs/installation/kubernetes/) on the cluster. Then, [configure](https://cert-manager.io/docs/configuration/) an Issuer/ClusterIssuer resource and last configure a Certificate resource to manage allocator-tls Secret.
```
Comment (aLekSer) [inline site/content/en/docs/Advanced/allocator-service.md:38]: nit: I find it useful to mark resources as code block as stated here https://developers.google.com/style/code-in-text and used in other places through out the `site/`
```suggestion
In order to use cert-manager solution, first, [install cert-manager](https://cert-manager.io/docs/installation/kubernetes/) on the cluster. Then, [configure](https://cert-manager.io/docs/configuration/) an `Issuer`/`ClusterIssuer` resource and last configure a `Certificate` resource to manage allocator-tls `Secret`. 
```

File: site/content/en/docs/Advanced/allocator-service.md (hunk 3 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md]: Thanks. Done.

File: site/content/en/docs/Advanced/allocator-service.md (hunk 4 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md]: Good suggestion. Thanks.

File: site/content/en/docs/Advanced/allocator-service.md [hunk 5]
```diff
@@ -33,36 +33,50 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
...
+  commonName: ${EXTERNAL_IP}
+  ipAddresses:
+    - ${EXTERNAL_IP}
+  secretName: allocator-tls
+  issuerRef:
+    name: selfsigned
+    kind: ClusterIssuer
+EOF
+
+# Optional: Store the secret ca.crt in a file to be used by the client for the server authentication
```
Comment (markmandel) [inline site/content/en/docs/Advanced/allocator-service.md:73]: Perfect!

**Comment by markmandel** [comment]
I'm not sure - I'm happy to approve this, but should it be merged before the stable comes out next week or after?

I'm thinking after, but I'm open to other arguments?

**Comment by pooneh-m** [comment]
Yes it should be release as it provides a better suggestion on managing server certificates on the cluster vs one time issuing and using a certificate. In the future, we may integrate the solution into Agones, according to #1461 if it makes sense.

**Pull request merged**